lib/rpm_package.py

```diff
@@ -96,9 +98,21 @@ def query_tag(self, tag_name, extra_args=""):
             tag_value = self._cached_tags[tag_name]
         else:
             extra_args += get_define_line(self._additional_macros)
+
+            # if a macro happens to not be defined, we can't find its
+            # location in the string after expansion, so we define a
+            # dummy value here that can be searched for later
+            dummy_macros = {m : "dummy_" + m for m in unexpanded_macros}
+            extra_args += get_define_line(dummy_macros)
+
             command = "rpmspec --srpm -q --qf '%%{%s}' %s %s 2>/dev/null" % (
                     tag_name.upper(), self.path, extra_args)
             tag_value = utils.run_command(command).strip()
+
+            for macro in unexpanded_macros:
+                tag_value = tag_value.replace('dummy_%s' % macro,
+                                              '%%{%s}' % macro)
```

Comment:
Just a thought: if the macro happens to be defined in the spec file, this replace will not work, since the dummy value will be overwritten by whatever is in the file. I don't think we have to worry with this use case though.